### PR TITLE
Update instructions and versions

### PR DESCRIPTION
**Pull request by angristan**
- Update Node LTS version and NPM.
- Update Mongo EOL warning
- Update Mongo and Redis version
- Update NodeBB branch
- I removed the Nginx version because there is no need for a specific one. Users should just install the latest stable one from their package manager. I also removed the Ubuntu PPA for Nginx. We should avoid PPA when not needed.
- Use the `root` role for `admin`, so that it will be able to remove DBs etc
- Bump Ubuntu version to latest LTS

### CODE REVIEW

File: src/templates/installing/os/template.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ Switch to the built-in `admin` database:
 Create an administrative user (the is different from the `nodebb` user we'll create later). Replace the placeholder `<Enter a secure password>` with your own selected password. Be sure that the `<` and `>` are also not left behind.
 
 ```
-> db.createUser( { user: "admin", pwd: "<Enter a secure password>", roles: [ { role: "readWriteAnyDatabase", db: "admin" }, { role: "userAdminAnyDatabase", db: "admin" } ] } )
+> db.createUser( { user: "admin", pwd: "<Enter a secure password>", roles: [ { role: "root", db: "admin" } ] } )
```
[the remark this answers is not in the document]
Comment (angristan) [reply src/templates/installing/os/template.md]: You need to be root to do some actions like delete a database. Since we give theses permissions to the admin and these enable `authorization`, there is no way to delete databases without having a root user. (from what I understood by having the issue while following the guide)

[the remark this answers is not in the document]
Comment (pitaj) [reply src/templates/installing/os/template.md]: This database user isn't used by NodeBB, so I don't think it really matters

[the remark this answers is not in the document]
Comment (julianlam) [reply src/templates/installing/os/template.md]: Oh I understand, that makes sense.

File: src/templates/installing/os/template.md [hunk 2]
```diff
@@ -153,7 +153,6 @@ Verify the installation of nginx
 
 ```bash
 {{commandPrefix}} nginx -v
-nginx version: nginx/{{versions.recommended.nginx}}
```
Comment (pitaj) [inline src/templates/installing/os/template.md]: Why is this removed?

File: src/templates/installing/os/ubuntu/nginx/install.md [hunk 3]
```diff
@@ -1,7 +1,3 @@
-The following is an abbreviation of the [Ubuntu PPA nginx installation instructions](https://www.nginx.com/resources/wiki/start/topics/tutorials/install/#ubuntu-ppa).
```
Comment (pitaj) [inline src/templates/installing/os/ubuntu/nginx/install.md]: Why remove this? Should people not be installing the latest stable nginx?

It's probably better to use the ppa since people could be on older versions of Ubuntu.

File: src/templates/installing/os/windows/nginx/install.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 The following is an abbreviation of the [Windows nginx installation instructions](https://www.nginx.com/resources/wiki/start/topics/tutorials/install/#official-win32-binaries).
 
 1. Create a new directory `C:\nginx`
-2. [Download the stable nginx v{{versions.recommended.nginx}} win32 binary](http://nginx.org/en/download.html)
```
Comment (pitaj) [inline src/templates/installing/os/windows/nginx/install.md]: Why remove the recommended nginx version? Maybe it would be better to automatically pull it from the internet like we do with the node version?

File: src/templates/installing/os/template.md (hunk 2 above)
Comment (angristan) [reply src/templates/installing/os/template.md]: See my initial comment on the PR.

File: src/templates/installing/os/ubuntu/nginx/install.md (hunk 3 above)
Comment (angristan) [reply src/templates/installing/os/ubuntu/nginx/install.md]: See my initial comment on the PR.

File: src/templates/installing/os/windows/nginx/install.md (hunk 4 above)
Comment (angristan) [reply src/templates/installing/os/windows/nginx/install.md]: See my initial comment on the PR.

**Comment by julianlam** [comment]
LGTM, although unfortunately it seems like my latest changes to change the recommended version to Ubuntu 18.04 may have introduced some conflicts... can you resolve?

**Comment by julianlam** [comment]
Ah I resolved interactively, although I changed the recommended redis version to 5.0.3, to match what `apt-cache policy` reports on my local dev machine for `redis-server`.

Any particular reason you want to set it to 5.4?

**Comment by angristan** [comment]
That probably a mistake, the latest version is indeed 5.0.3, 5.4 does not exist 😄

Maybe we should recommend just "5" to no be too specific? By the way the Redis version on Ubuntu 18.04 is 4.0.9, and 3.2.6 on Debian 9, so maybe we should say 4 ? or >=3 ? 

I don't know 🤷‍♂️I think 5.0.3 is too recent and too specific (and so was 5.4 😉)

**Comment by julianlam** [comment]
Ah you are correct, it turns out I _did_ have the chris-lea PPA installed, whoops. I am perfectly find with adjusting this to `4.0.9`. So long as it's higher than the minimum version, NodeBB should be fine...

**Comment by angristan** [comment]
Yep 👍
